lib/puppet/type/firewall.rb

```diff
@@ -780,6 +780,33 @@ def insync?(is)
     EOS
   end
 
+  # match mark
+  newproperty(:match_mark, :required_features => :mark) do
+    desc <<-EOS
+      Match the Netfilter mark value associated with the packet.  Accepts either of:
+      mark/mask or mark.  These will be converted to hex if they are not already.
+    EOS
+    munge do |value|
+      int_or_hex = '[a-fA-F0-9x]'
+      match = value.to_s.match("(#{int_or_hex}+)(/)?(#{int_or_hex}+)?")
```

Comment:
A better matcher would be:
```ruby
mark_regex = %r{\A((?:0x)?[0-9A-F]+)(?:/((?:0x)?[0-9A-F]+))?\z}I
match = value.to_s.match(mark_regex)
```
This has the added benefit of requiring significantly less processing of the match object.